test/performance/kperf_helper.go

```diff
@@ -0,0 +1,154 @@
+//go:build performance
+// +build performance
+
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package performance
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	networkingv1alpha1 "knative.dev/networking/pkg/client/clientset/versioned/typed/networking/v1alpha1"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/injection"
+	pkgTest "knative.dev/pkg/test"
+	"knative.dev/pkg/test/logging"
+	"knative.dev/serving/pkg/client/clientset/versioned"
+	autoscalingv1alpha1 "knative.dev/serving/pkg/client/clientset/versioned/typed/autoscaling/v1alpha1"
+	servingv1client "knative.dev/serving/pkg/client/clientset/versioned/typed/serving/v1"
+	"knative.dev/serving/test"
+)
+
+// Setup creates client to run Knative Service requests
+func testSetup(ctx context.Context, t testing.TB, namespace ...string) *test.Clients {
```

Comment:
The "namespace" param looks unused. Should this be passed to `test.NewClients()` instead of the default namespace? Then the function servingClientForNamespace could just return `ServingClient` from test.Clients struct instead of initializing the whole serving client again.

Reply:
will switch over to that